src/mount/writedata.cc

```diff
@@ -256,42 +255,31 @@ void write_cb_wait_for_block(inodedata* id, Glock& glock) {
 
 /* inode */
 
-inodedata* write_find_inodedata(uint32_t inode, Glock&) {
-	uint32_t idh = IDHASH(inode);
-	for (inodedata* id = idhash[idh]; id; id = id->next) {
-		if (id->inode == inode) {
-			return id;
-		}
+inodedata *write_find_inodedata(uint32_t inode, Glock &) {
+	if (inodedataMap.contains(inode)) {
+		return inodedataMap[inode];
 	}
 	return NULL;
 }
 
-inodedata* write_get_inodedata(uint32_t inode, Glock&) {
-	uint32_t idh = IDHASH(inode);
-	inodedata* id;
-	for (inodedata* id = idhash[idh]; id; id = id->next) {
-		if (id->inode == inode) {
-			return id;
-		}
+inodedata *write_get_inodedata(uint32_t inode, Glock &) {
+	if (inodedataMap.contains(inode)) {
+		return inodedataMap[inode];
 	}
-	id = new inodedata(inode);
-	id->next = idhash[idh];
-	idhash[idh] = id;
+	inodedata *id = new inodedata(inode);
```

Comment:
Consider using `auto` here to avoid duplicating `inodedata` type.

Reply:
Applied.